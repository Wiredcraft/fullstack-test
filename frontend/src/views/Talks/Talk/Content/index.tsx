import React, {ReactElement} from 'react';
import {ITalkObject} from 'src/interfaces/ITalk';

interface ContainerProps {
    children?: React.ReactNode;
    talk: ITalkObject;
}

const TalkContent = (props: ContainerProps): ReactElement => {
  return (
    <div style={{
      flex: '1 0 auto',
      margin: '10px',
      alignContent: 'center',
      alignItems: 'center',
      textAlign: 'left',
      flexBasis: '40%',
    }}>
          Name: {props.talk.name}
      <br/>
          Description: {props.talk.description}
      <br/>
          Author: {props.talk.user}
      <br/>
          Submission time: {new Date(props.talk.createdAt).toUTCString()}
    </div>
  );
};

export default TalkContent;
