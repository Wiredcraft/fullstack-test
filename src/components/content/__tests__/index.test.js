import React from 'react';
import renderer, { act } from 'react-test-renderer';
import wait from 'waait';

import Content from '../index';

jest.mock('react-router-dom');
jest.mock('axios');
jest.mock('../../../services/topics', () => ({
  getTopics: () => []
}));

test('Content renders', async () => {
  let component;

  await act(async () => {
    component = renderer.create(<Content />);
    await wait();
  });

  expect(component.toJSON()).toMatchSnapshot();
});
