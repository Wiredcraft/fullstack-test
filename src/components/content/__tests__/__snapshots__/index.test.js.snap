// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders 1`] = `null`;

exports[`Content renders with empty topics 1`] = `
<div
  className="sc-bdvvtL cdNifx"
>
  <p
    className="sc-gsDKAQ liIgxQ"
  >
    No talks yet... Be the first!
  </p>
  <button
    className="sc-dkPtRN bIiLjj"
    onClick={[Function]}
  >
    Add your lightning talk
  </button>
</div>
`;

exports[`Content renders with populated topics 1`] = `
<table
  className="sc-iqseJM deCyjf"
>
  <tbody>
    <tr
      className="sc-hKwDye lnnZFs"
    >
      <td
        className="sc-eCImPb hciDOa"
      >
        <div
          className="sc-furwcr dszeXW"
        >
          <div
            className="sc-jrQzAO dipdKg"
            onClick={[Function]}
          />
          <div
            className="sc-kDTinF jojDFf"
            onClick={[Function]}
          />
          <span
            className="sc-pVTFL gNSwkQ"
          />
        </div>
      </td>
      <td
        className="sc-jRQBWg fUobTI"
        onClick={[Function]}
      >
        <div
          className="sc-gKclnd kRMSpt"
        />
        <div
          className="sc-iCfMLu BuLQm"
        >
          by 
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
